api/Document.json

```diff
@@ -623,10 +623,10 @@
               "version_added": "12"
             },
             "firefox": {
-              "version_added": true
+              "version_added": "51"
```

Comment:
This does match results for http://mdn-bcd-collector.appspot.com/tests/api/Document/onanimationend and https://mdn-bcd-collector.appspot.com/tests/api/HTMLElement/onanimationend but still seems way too late.

`AnimationEvent` is marked as shipped in Firefox 6 and the `animation` CSS property unprefixed in 16. Then why no "animationend" events until that much later? Can you do some manual testing of when the event is fired?

Reply:
I've given these a manual test in older Firefox editions.  It turns out that there was a big difference between when the events were supported vs. the event handlers.  I've updated the data using the following tests (which is a modified version of the test on https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/animationend_event for compatibility with older browser versions):

```html
<head>
		<!-- <meta content="width=device-width"> -->
		<style id="persistent-style">
			/* Persistent styling */
			body {
				background-color: black;
				color: white;
				margin: 0;
				padding: 0;
			}
			canvas {
				/*background-color: white;*/
			}
		</style>
		<style id="test-style">
			.container {
			  height: 3rem;
			}

			.event-log {
			  width: 25rem;
			  height: 2rem;
			  border: 1px solid black;
			  margin: .2rem;
			  padding: .2rem;
			}

			.animation.active {
			  -moz-animation-duration: 2s;
			  -moz-animation-name: slidein;
			  -moz-animation-iteration-count: 2;
			}

			@-moz-keyframes slidein {
			  from {
			    margin-left: 100%;
			    width: 300%;
			  }

			  to {
			    margin-left: 0%;
			    width: 100%;
			  }
			}
		</style>
	</head>
	<body>
		<div id="test">
			<div class="animation-example">
		    <div class="container">
		        <p class="animation">You chose a cold night to visit our planet.</p>
		    </div>
		    <button class="activate" type="button">Activate animation</button>
		    <div class="event-log"></div>
			</div>
		</div>

		<script>
			var animation = document.querySelector('p.animation');
			var animationEventLog = document.querySelector('.animation-example>.event-log');
			var applyAnimation = document.querySelector('.animation-example>button.activate');
			var iterationCount = 0;

			animation.addEventListener('animationstart', function() {
			  animationEventLog.textContent = 'animation started' ;
			});

			animation.addEventListener('animationiteration', function() {
			  iterationCount++;
			  animationEventLog.textContent = 'animation iterations: ' + iterationCount;
			});

			animation.addEventListener('animationend', function() {
			  animationEventLog.textContent = 'animation ended';
			  animation.classList.remove('active');
			  applyAnimation.textContent = "Activate animation";
			});

			animation.addEventListener('animationcancel', function() {
			  animationEventLog.textContent = 'animation canceled';
			});

			applyAnimation.addEventListener('click', function() {
			  animation.classList.toggle('active');
			  animationEventLog.textContent = '';
			  iterationCount = 0;
			  var active = animation.classList.contains('active');
			  if (active) {
			    applyAnimation.textContent = "Cancel animation";
			  } else {
			    applyAnimation.textContent = "Activate animation";
			  }
			});
		</script>
	</body>
```